commands/deploy.go

```diff
@@ -249,7 +250,34 @@ Error: %s`, fprocessErr.Error())
 				function.ReadOnlyRootFilesystem = deployFlags.readOnlyRootFilesystem
 			}
 
-			statusCode := proxy.DeployFunction(function.FProcess, services.Provider.GatewayURL, function.Name, function.Image, function.RegistryAuth, function.Language, deployFlags.replace, allEnvironment, services.Provider.Network, functionConstraints, deployFlags.update, deployFlags.secrets, allLabels, annotations, functionResourceRequest1, function.ReadOnlyRootFilesystem, tlsInsecure)
+			// Need to alter Gateway to allow nil/empty string as fprocess, to avoid this repetition.
```

Comment:
I kept this condition while moving the code out of `proxy_deploy.go`. Is this check needed?

Reply:
From what I've seen, it's basically a `nil` / empty check for the function process. But the zero-value of strings are empty, so I don't see how this is benefiting. Please let me know if I'm missing anything.

Reply:
I agree I don't see what error the code is catching.

Reply:
I can't see the line, in GitHub it shows you highlighted a comment.

Reply:
@alexellis https://github.com/james-r-smith/faas-cli/blob/296d43bc2ad666bd37d60e60ceb514e4a8bb4a7f/commands/deploy.go#L253-L257

I'm not sure how to comment on a block. 😅